### DISCUSSION TITLE
how to use equivalent of `flushSync` ?

### DISCUSSION THREAD

**Discussion by arnaudambro** [facebookexperimental/Recoil#1879] [Q&A]
Hello

I have a callback where I need to update an atom then use the new value of it.
It looks like

```jsx
const [migrations, setMigrations] = useRecoilState(migrationsState)
const doMigrations = async () => {
  if (!migrations.includes('migration-1')) {
    // do some stuff
    setMigrations(m => [...m, 'migration-1'])
  }

  /* 
  NOW I NEED THE `migrations` to includes 'migration-1' before doing other stuff
  */

  console.log(migrations.includes('migration-1')) // false

  if (!migrations.includes('migration-2')) {
    // do some stuff
    setMigrations(m => [...m, 'migration-2'])
  }
}
```

in React, there is the `flushSync` that would do the thing, but I can't find out how to do achieve this in recoil

Is there a way ?

**Comment by drarmstr** [comment]
For atomic transactions you can use `useRecoilTransaction_UNSTABLE()`.  The transaction will observe previous mutations.  However, there are current limitations that only sync operations with atoms are supported.  For more complete support you can use `useRecoilCallback()` which is provided a snapshot.
Snapshots are immutable, but you can transform it with `map()`, observe previous mutations as you go, and then goto the snapshot.  If it is async, though, be careful of intervening operations.
